### PR TITLE
Added endpoint to get QR code string

### PR DESCRIPTION
**Pull request by PradyumnaShome**
Addresses #1 

 - [x] Added an endpoint to get the string which should be embedded into the user's QR code
    - `GET /checkin/qr/`

The QR code string contains
- [x] User ID
- [x] Check-in status
- [x] Swag collection status

### CODE REVIEW

**Comment by PradyumnaShome** [comment]
Should I write a service level test for this?

File: service/checkin_service.go [hunk 1]
```diff
@@ -69,3 +71,40 @@ func UpdateUserCheckin(id string, user_checkin models.UserCheckin) error {
...
+	
+	checkin_status, err := GetUserCheckin(id)
+
+	if err != nil {
+		return "", err
+	}
+
+	has_checked_in_string := strconv.FormatBool(checkin_status.HasCheckedIn)
+
+	has_picked_up_swag_string := strconv.FormatBool(checkin_status.HasPickedUpSwag)
```
Comment (ASankaran) [inline service/checkin_service.go]: These are simple conversions. It's easier to just inline them down where you place them into the uri parameters

File: models/qr_info_container.go [hunk 2]
```diff
@@ -0,0 +1,6 @@
+package models
+
+type QrInfoContainer struct {
+	ID 		 string `json:"id"`
+	QrInfo   string `json:"qrInfo"`
```
Comment (ASankaran) [inline models/qr_info_container.go]: Did you run `go fmt ./...` on the repo? It should align the types.

File: tests/checkin_test.go [hunk 3]
```diff
@@ -150,3 +150,24 @@ func TestUpdateUserCheckinService(t *testing.T) {
...
+	
+	actual_uri, err := service.GetQrInfo("testid")
+	
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	expected_uri := "hackillinois://info?hasCheckedIn=true&hasPickedUpSwag=true&userId=testid"
+
+	if !reflect.DeepEqual(actual_uri, expected_uri) {
```
Comment (ASankaran) [inline tests/checkin_test.go]: Decode the uri to a map of query parameter and do a `DeepEqual ` against a map of the expected parameters.

File: docs/Checkin.md [hunk 4]
```diff
@@ -74,3 +74,31 @@ Response format:
...
+----------
+
+Get the string to be embedded in the current user's QR code. 
+The QR code string will contain information stored in the form of a URI.
+
+Response format:
+```
+{
+	"id": "github0000001",
+	"qrInfo": "hackillinois://info?name=FirstName%20LastName&hasCheckedIn=true"
```
Comment (ASankaran) [inline docs/Checkin.md]: These parameters differ from what the endpoint actually returns.

File: docs/Checkin.md [hunk 5]
```diff
@@ -74,3 +74,31 @@ Response format:
...
+----------
+
+Get the string to be embedded in the specified user's QR code. 
+The QR code string will contain information stored in the form of a URI.
+
+Response format:
+```
+{
+	"id": "github0000001",
+	"qrInfo": "hackillinois://info?name=FirstName%20LastName&hasCheckedIn=true"
```
Comment (ASankaran) [inline docs/Checkin.md]: These parameters differ from what the endpoint actually returns.

File: models/qr_info_container.go (hunk 2 above)
Comment (PradyumnaShome) [reply models/qr_info_container.go]: Will do. I think I used Tab accidentally.

File: docs/Checkin.md (hunk 4 above)
Comment (PradyumnaShome) [reply docs/Checkin.md]: I thought you said it was implementation-dependent?

Comment (ASankaran) [reply docs/Checkin.md]: The ordering of the parameters is implementation dependent, but here you have a uri containing `name` and `hasCheckIn` parameters. The endpoint returns a uri with `userId`, `hasPickedUpSwag`, and `hasCheckedIn` parameters. The set of parameters in the docs needs to match the endpoint, but the order is not defined.

Comment (PradyumnaShome) [reply docs/Checkin.md]: Sure.

**Pull request merged**